contracts/shared_traits/src/lib.rs

```diff
@@ -0,0 +1,5 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+mod haltable;
+
+pub use haltable::*;
```

Comment:
nit: This could be just
```rust
pub mod haltable;
```
and then on the use-side:
```
use shared_traits::haltable::{DefaultHaltable, Haltable}
```
instead of just
```rust
use shared_traits::Haltable
```
This looks as better package management that groups functionality together. If we add more things into `shared_traits` that are not haltable-related then it will become its own package.

Reply:
I'm following https://www.notion.so/cardinalcryptography/Module-based-access-in-our-codebase-b08edefff9394f68938da09030c0bda6?pvs=4#4a539c6232e144b49e03ef94a6aa5e78

Reply:
Different thing - the notion ADR is about what and how to share. I'm talking about grouping the elements behind the logical modules. You could still achieve both here, for example by re-rexporting:
```rust
mod haltable_internal; // or whatever
pub mod haltable {
  use haltable_internal::{...}
}
```
which would still make it available through `use shared_traits::haltable::*`.

Not a blocker but I think we would eventually follow-up with something similar, like we did for all other repos eventually.

Reply:
This is literally written down as a no-go in the ADR:
```
All modules are private (no pub mod foo), if we want something from a submodule to be accessible from outside of the module then we have to explicitly reexport it (e.g. pub use foo::Bar in mod.rs).
```
I don't care what the convention is as long as it's consistent. I don't see a point in discussing it here - the ADR can be re-visisted if it really itches someone.

Reply:
You still don't understand - these are two different things:
1. Not making whole module `pub`
2. Re-exporting things from a module with a wrapper.
I'm not asking for the 1), but 2). `pub use foo::Bar` makes `Bar` being accessed as a top-level element:
```rust
mod foo {
    // I want this public.
    pub struct Foo;
    
    // And this private.
    pub struct FooPriv;
    
}

pub use foo::Foo;

mod bar {
    use crate::Foo;
}
```
vs
```rust
mod foo_internal {
    // I want this public.
    pub struct Foo;
    // And this private.    
    pub struct FooPriv;
}

pub mod foo_public {
    pub use crate::foo_internal::Foo;
}

use foo_public::Foo;
```

In the first version `Foo` is accessed directly from the top-level module while in the latter is behind logicaly-grouped `foo` package (in the case of this PR it would be `use shared_traits::haltable::Haltable` vs `use shared_traits::Haltable`).

It's completely independent from the ADR that you linked.